d3a_api_client/redis_client_base.py

```diff
@@ -2,57 +2,73 @@
 import logging
 import uuid
 from concurrent.futures.thread import ThreadPoolExecutor
-from functools import wraps
 
 from d3a_interface.utils import wait_until_timeout_blocking, key_in_dict_and_not_none
 from redis import StrictRedis
+from slugify import slugify
 
 from d3a_api_client import APIClientInterface
 from d3a_api_client.constants import MAX_WORKER_THREADS
+from d3a_api_client.utils import execute_function_util
 
 
 class RedisAPIException(Exception):
     pass
 
 
-def registered_connection(f):
-    @wraps(f)
-    def wrapped(self, *args, **kwargs):
-        if not self.is_active:
-            raise RedisAPIException(f'Registration has not completed yet.')
-        return f(self, *args, **kwargs)
-
-    return wrapped
-
-
-class RedisClient(APIClientInterface):
+class RedisClientBase(APIClientInterface):
     def __init__(self, area_id, autoregister=True, redis_url='redis://localhost:6379',
                  pubsub_thread=None):
         super().__init__(area_id, autoregister, redis_url)
+        self.area_uuid = None
+        self.area_slug = slugify(area_id, to_lower=True)
         self.redis_db = StrictRedis.from_url(redis_url)
         self.pubsub = self.redis_db.pubsub() if pubsub_thread is None else pubsub_thread
         self.area_id = area_id
         self.device_uuid = None
         self.is_active = False
         self._blocking_command_responses = {}
+        self._transaction_id_buffer = []
+        self._subscribed_aggregator_response_cb = None
         self._subscribe_to_response_channels(pubsub_thread)
         self.executor = ThreadPoolExecutor(max_workers=MAX_WORKER_THREADS)
         if autoregister:
             self.register(is_blocking=True)
 
+    @property
+    def _channel_prefix(self):
+        return f"{self.area_slug}"
+
```

Comment:
I have the feeling this method might be removed since it's just returning `self.area_slug`. The f-string here shouldn't be needed in any case, if slugify is returning a string. What do you think?

Reply:
You are right, will remove it and will use the area_slug in the channel_subs function directly.